File: py_validation/pardox/_bindings/strings.py
def bind(lib, c_void_p, c_char_p, c_int32, c_double, c_longlong, c_size_t, c_int64, c_int16):
    for fn, args, ret in [
        ('pardox_str_upper',    [c_void_p, c_char_p],                          c_void_p),
        ('pardox_str_lower',    [c_void_p, c_char_p],                          c_void_p),
        ('pardox_str_len',      [c_void_p, c_char_p],                          c_void_p),
        ('pardox_str_trim',     [c_void_p, c_char_p],                          c_void_p),
        ('pardox_str_replace',  [c_void_p, c_char_p, c_char_p, c_char_p],     c_void_p),
        ('pardox_str_contains', [c_void_p, c_char_p, c_char_p],               c_void_p),
    ]:
        if hasattr(lib, fn):
            getattr(lib, fn).argtypes = args
            getattr(lib, fn).restype  = ret

File: js_validation/pardox/src/_bindings/strings.js
'use strict';

module.exports = function bindStrings(_lib) {
    _lib.pardox_str_upper    = _lib.raw.func('void * pardox_str_upper(void * mgr, const char * col_name)');
    _lib.pardox_str_lower    = _lib.raw.func('void * pardox_str_lower(void * mgr, const char * col_name)');
    _lib.pardox_str_contains = _lib.raw.func('void * pardox_str_contains(void * mgr, const char * col_name, const char * pattern)');
    _lib.pardox_str_replace  = _lib.raw.func('void * pardox_str_replace(void * mgr, const char * col_name, const char * old_str, const char * new_str)');
    _lib.pardox_str_trim     = _lib.raw.func('void * pardox_str_trim(void * mgr, const char * col_name)');
    _lib.pardox_str_len      = _lib.raw.func('void * pardox_str_len(void * mgr, const char * col_name)');
};
